plan/physical_plan_builder.go

```diff
@@ -245,7 +245,11 @@ func (p *DataSource) convert2IndexScan(prop *requiredProperty, index *model.Inde
 		if client != nil {
 			memDB := infoschema.IsMemoryDB(p.DBName.L)
 			if !memDB && client.SupportRequestType(kv.ReqTypeIndex, 0) {
-				is.ConditionPBExpr, is.conditions, newSel.Conditions = expressionsToPB(sc, newSel.Conditions, client)
+				var indexConditions []expression.Expression
+				indexConditions, newSel.Conditions = detachIndexFilterConditions(newSel.Conditions, is.Index.Columns, is.Table)
```

Comment:
Define a new temporary variable `tableConditons` instead of use `newSel.Conditon` is better.